### ISSUE TITLE
`/usr/local/cuda/bin` is not added to `$PATH`

### ISSUE THREAD

**Issue by leofang** [conda-forge/docker-images#184]
It appears that #181 did not fix the problem, see https://github.com/conda-forge/cutensor-feedstock/pull/12#issuecomment-853318652.

**Comment by leofang** [comment]
Looks like the aarch64 images have the same problem (https://github.com/conda-forge/docker-images/pull/189).

cc: @jakirkham (for awareness)

**Comment by jakirkham** [comment]
For the PPC images, PR ( https://github.com/conda-forge/docker-images/pull/181 ) added this

https://github.com/conda-forge/docker-images/blob/91576a9507829f2a28b2ce5c5d7f39d59afb22b9/linux-anvil-ppc64le-cuda/entrypoint_source#L1-L2

<hr>

That said, ARM images currently lack those lines

**Comment by jakirkham** [comment]
cc @jaimergp (in case you have thoughts on this as well)

**Comment by jakirkham** [comment]
Actually I think this has to do without how we run with QEMU. Notice with [`x86_64` we add `-it` to `docker run`]( https://github.com/conda-forge/cutensor-feedstock/blob/6221d580efcadfd89da65a72a265861f1aaadc89/.scripts/run_docker_build.sh#L69-L71 ), but when using [QEMU we don't pass the `CI` environment variable]( https://github.com/conda-forge/cutensor-feedstock/blob/150be5793bca80ec8f87ee95fadd95aeb24cd1dc/.azure-pipelines/azure-pipelines-linux.yml#L42 ). So that check likely fails resulting in `-it` not being passed to `docker run`

**Comment by jakirkham** [comment]
I'm wondering if this has something to do with how we are running QEMU. Normally when calling `docker run`, we add [`-it`]( https://github.com/conda-forge/cutensor-feedstock/blob/150be5793bca80ec8f87ee95fadd95aeb24cd1dc/.scripts/run_docker_build.sh#L69-L71 ), which should spin up a pseudo-TTY in interactive mode. When that wasn't done in the past, the source activation wouldn't work correctly.

**Comment by jaimergp** [comment]
That's weird indeed and would explain why we still needed to manually export CUDA_PATH on the ppc builds for OpenMM, iirc.

**Comment by jakirkham** [comment]
cc @mariusvniekerk (in case you have insights on how we might get this working with QEMU 🙂)

**Comment by jakirkham** [comment]
Turns out the issue is pretty simple actually. We weren't getting the right entrypoint script. Fixing with PR ( https://github.com/conda-forge/docker-images/pull/210 ).

That said, we still have build issues with the CUDA arch images ( https://github.com/conda-forge/docker-images/issues/208 ). So will need to sort that out somehow to get new image builds.

**Comment by leofang** [comment]
Good catch! Thanks, John.

**Comment by jakirkham** [comment]
Ofc images are building (though pretty close to done) so may take a bit more before the fix is deployed

**Comment by jakirkham** [comment]
All of the images were built and pushed except `conda-forge/linux-anvil-ppc64le-cuda:11.6`, which ran into some timeout downloading thing. Restarted just that build.

**Comment by jakirkham** [comment]
Submitted a couple PRs to drop the `CUDA_HOME` workaround:

* https://github.com/conda-forge/cub-feedstock/pull/21
* https://github.com/conda-forge/thrust-feedstock/pull/16

Though haven't handled all cases. This is probably a good starting point for [a list of them]( https://github.com/search?l=YAML&q=org%3Aconda-forge+CUDA_HOME&type=Code ).

**Comment by leofang** [comment]
Yeah I'll drop them gradually when working on new PRs. I don't wanna send a PR just to drop it.

**Comment by jakirkham** [comment]
Yep it's not urgent. Mostly opened those to verify things are now working with the added bonus of doing some cleanup.
